core-web/libs/dotcms-webcomponents/src/components/dot-html-to-image/dot-html-to-image.tsx

```diff
@@ -0,0 +1,108 @@
+import { Component, Prop, h, Host, Event, EventEmitter, State } from '@stencil/core';
+import '@material/mwc-circular-progress';
+
+@Component({
+    tag: 'dot-html-to-image',
+    styleUrl: 'dot-html-to-image.scss',
+    shadow: false
+})
+export class DotHtmlToImage {
+    @Prop({ reflect: false })
+    value = '';
+
+    @Prop({ reflect: false, mutable: true })
+    height = '';
+
+    @Prop({ reflect: false, mutable: true })
+    width = '';
+
+    @Event() pageThumbnail: EventEmitter<{
+        file: File;
+        error?: string;
+    }>;
+    @State() previewImg: string;
+
+    iframeId = `iframe_${Math.floor(Date.now() / 1000).toString()}`;
+    loadScript = `
+        html2canvas(document.body, {
+            height: IMG_HEIGHT, // The height of the canvas
+            windowHeight: IMG_HEIGHT, // Window height to use when rendering Element
+            width: IMG_WIDTH, // The width of the canvas
+            windowWidth: IMG_WIDTH, // Window width to use when rendering Element
+        }).then((canvas) => {
+            canvas.toBlob((blob) => {
+                const fileObj = new File([blob], 'image.png');
+                window.parent.postMessage({
+                    iframeId: '${this.iframeId}',
+                    previewImg: canvas.toDataURL(),
+                    fileObj
+                }, '*')
+            });
+        }).catch(function (error) {
+            window.parent.postMessage({
+                iframeId: '${this.iframeId}',
+                error
+            }, '*')
+        });
+    ;`;
+
+    componentDidLoad() {
+        const iframe = document.querySelector(`#${this.iframeId}`) as HTMLIFrameElement;
+        const doc = iframe.contentDocument || iframe.contentWindow.document;
+
+        doc.open();
+        doc.write(this.value);
+        doc.close();
+
+        const scriptLib = document.createElement('script') as HTMLScriptElement;
+        scriptLib.src = '/html/js/html2canvas/html2canvas.min.js';
+        scriptLib.type = 'text/javascript';
+
+        scriptLib.onload = () => {
+            iframe.addEventListener('load', () => {
+                const script: HTMLScriptElement = document.createElement('script');
+                script.type = 'text/javascript';
+                script.text = this.width
+                    ? this.loadScript
+                          .replace(/IMG_HEIGHT/g, this.height)
+                          .replace(/IMG_WIDTH/g, this.width)
+                    : this.loadScript;
+                doc.body.appendChild(script);
+            });
+        };
+
+        doc.body.append(scriptLib);
+
+        window.addEventListener('message', (event) => {
```

Comment:
Take a look at this video: https://youtu.be/YDU_3WdfkxA?t=681

I think this code will result in the same issue.

Reply:
fixed!